src/pseudopeople/interface.py

```diff
@@ -77,6 +77,12 @@ def _generate_dataset(
         noised_data = _extract_columns(dataset.columns, noised_data)
         noised_dataset.append(noised_data)
 
+    # Check if all shards for the dataset are empty
+    if len(noised_dataset) == 0:
+        raise ValueError(
+            "Invalid value provided for 'state' or 'year'. User input value(s) provided "
+            f"are {user_filters}. No data found with these filters at {data_path}."
```

Comment:
Question about `Invalid value provided for 'state' or 'year'.` Seems like an assumption here where we could validate that quickly before attempting to noise? Also, are these the only two possibilities (i.e., it's unlikely/impossible to noise your data so pathologically that it's empty)?

Reply:
If the user loads a dataframe from a file and it is empty (whether from user filters or not) it will not be noised. It is technically possible that we have a very very small dataset that gets loaded and then we omit the very few rows. That is the only use case I have been able to think of.

Reply:
Note that even though this says `len(noised_dataset)`, this is _before_ the `pd.concat`, so this checks whether any non-empty dataframes were loaded, not whether any rows still remain after noise. That's correct -- we don't want this error if the empty dataframe is due to noise, only if it is due to there being no data in the first place.